source/_integrations/private_ble_device.markdown

```diff
@@ -0,0 +1,43 @@
+---
+title: Private BLE Device
+description: Instructions on how to integrate private BLE devices into Home Assistant.
+ha_category:
+  - Device Tracker
+ha_release: '2023.10'
+ha_iot_class: Local Push
+ha_domain: private_ble_device
+ha_platforms:
+  - device_tracker
+ha_bluetooth: true
+ha_config_flow: true
+ha_integration_type: integration
+ha_codeowners:
+  - '@Jc2k'
+---
+
+Some BLE devices use a privacy feature called Resolvable Private Addresses to stop businesses tracking you when you are out with your smart devices. Instead of having a single static address that can be used to track you, its Bluetooth address changes frequently. If you have [Bluetooth](/integrations/bluetooth) enabled and functioning and you know the Identity Resolving Key for your device, you can add it to Home Assistant as a Private BLE Device. We can then work out which random mac addresses are related to the IRK and track those.
+
+{% include integrations/config_flow.md %}
+
+Your device must be on and broadcasting in range before adding it to Home Assistant - we check that it's visible to validat your IRK is correct.
+
+There are 2 common representations for encoding an IRK - base64 encoding or hex encoding. This integration supports both.
+
+## Getting your Identity Resolving Key
+
+### On macOS
+
+If you are trying to get the IRK for your iPhone or Apple Watch you'll need to be logged in to the Mac with the same iCloud account as on those devices. This should also work for devices that you pair with macOS.
+
+1. Start the **Keychain Acess** application.
+2. In the left sidebar make sure iCloud is selected.
+3. In the search bar in the upper right type Bluetooth.
+4. A list of GUID are shown.
+5. Double click on a record. As part of the **Account** field it will say `Public: XX:XX:XX:XX:XX:XX`. This is a MAC address and should match the device you are trying to work.
```

Comment:
```suggestion
5. Double-click on a record. As part of the **Account** field, it will say `Public: XX:XX:XX:XX:XX:XX`. This MAC address should match the device you are trying to work with.
```